ci/build_container/build_recipes/yaml-cpp.sh

```diff
@@ -4,11 +4,26 @@ set -e
 
 VERSION=0.6.2
 
-wget -O yaml-cpp-"$VERSION".tar.gz https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-"$VERSION".tar.gz
+curl https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-"$VERSION".tar.gz -sLo yaml-cpp-"$VERSION".tar.gz
 tar xf yaml-cpp-"$VERSION".tar.gz
 cd yaml-cpp-yaml-cpp-"$VERSION"
-cmake -DCMAKE_INSTALL_PREFIX:PATH="$THIRDPARTY_BUILD" \
+
+mkdir build
+cd build
+
+build_type=RelWithDebInfo
+if [[ "${OS}" == "Windows_NT" ]]; then
+  build_type=Debug
```

Comment:
Why do we have these debug exceptions for Windows everywhere?

Reply:
Envoy and all of its dependencies need to be compiled to use the same version of the C Runtime Library: https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library

If we build Envoy with `-c dbg`, it will use the Debug version of the C Runtime Library. We needed to set this option to force these dependencies to use the Debug library as well.

Right now, we've only got bazel building Envoy with `-c dbg` -- we haven't had a chance to get `fastbuild` or `opt` working. When we do that, there will probably have to be some more logic to select the appropriate build type

Reply:
Please then add some comments to this effect at these sites, thanks.